hugr-py/src/hugr/_hugr.py

```diff
@@ -337,17 +338,32 @@ def insert_hugr(self, hugr: Hugr, parent: Node | None = None) -> dict[Node, Node
             )
         return mapping
 
-    def add_dfg(self, input_types: Sequence[Type], output_types: Sequence[Type]) -> Dfg:
-        from ._dfg import Dfg
+    def add_dfg(self, root_op: DP) -> DfBase[DP]:
+        from ._dfg import DfBase
 
-        dfg = Dfg(input_types, output_types)
+        dfg = DfBase(root_op)
         mapping = self.insert_hugr(dfg.hugr, self.root)
         dfg.hugr = self
         dfg.input_node = mapping[dfg.input_node]
         dfg.output_node = mapping[dfg.output_node]
         dfg.root = mapping[dfg.root]
         return dfg
 
+    def add_cfg(self, input_types: Sequence[Type], output_types: Sequence[Type]) -> Cfg:
+        from ._cfg import Cfg
+
+        cfg = Cfg(input_types, output_types)
+        mapping = self.insert_hugr(cfg.hugr, self.root)
+        cfg.hugr = self
+        cfg._entry_block.root = mapping[cfg.entry]
+        cfg._entry_block.input_node = mapping[cfg._entry_block.input_node]
+        cfg._entry_block.output_node = mapping[cfg._entry_block.output_node]
+        cfg._entry_block.hugr = self
+        cfg.exit = mapping[cfg.exit]
+        cfg.root = mapping[cfg.root]
+        # TODO this is horrible
```

Comment:
It might be nicer to add an optional `hugr` argument to `Cfg.__init__`? If supplied, the CFG is inserted into the given Hugr, otherwise a new one is created.

Alternatively, it could be a classmethod `Cfg.new_nested(hugr, input_types, output_types)` or something similar?

Reply:
I like the class method idea, I'll try it out on #1194